modules/pinger/src/main/java/org/hawkular/component/pinger/Traits.java

```diff
@@ -0,0 +1,190 @@
+import com.google.common.collect.ImmutableMap.Builder;
+
+/**
+ * A collection of traits retrieved from response headers. The headers that are considered to be interesting are listed
+ * in {@link TraitHeader} enum. Trait is an interesting piece of information about a monitored site, such as the name of
+ * the web server implementation that is serving the site, the technology used in the serving application, etc.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class Traits {
+
+    /**
+     * The enumeration of HTTP response headers that are considered to bear interesting information about a monitored
+     * site.
+     */
+    public enum TraitHeader {
+        SERVER("server"), X_ASPNET_VERSION("x-aspnet-version"), X_POWERED_BY("x-powered-by"), X_RUNTIME("x-runtime"),
+        X_VERSION("x-version");
+
+        private static final Map<String, TraitHeader> index;
+        static {
+            TraitHeader[] traitHeaders = TraitHeader.values();
+            Map<String, TraitHeader> tmp = new HashMap<String, TraitHeader>(traitHeaders.length + traitHeaders.length
+                    / 2);
+            for (TraitHeader h : traitHeaders) {
+                tmp.put(h.toString(), h);
+            }
+            index = Collections.unmodifiableMap(tmp);
+        }
+
+        /**
+         * A case-insensitive and null-tolerant variant of {@link #valueOf(String)}.
+         *
+         * @param header
+         *            the header name to transform to a {@link TraitHeader}
+         * @return the corresponding {@link TraitHeader} or {@code null} if there si no such {@link TraitHeader}
+         */
+        public static TraitHeader fastValueOf(String header) {
+            return index.get(header.toLowerCase(Locale.US));
+        }
+
+        /** The name of the header */
+        private final String header;
+
+        private TraitHeader(String header) {
+            this.header = header.toLowerCase(Locale.US);
+        }
+
+        /** @return {@link #header} rather than {@link #name()} */
+        @Override
+        public String toString() {
+            return header;
+        }
+    }
+
+    /** An empty map to use for {@link #items} */
+    private static final Map<TraitHeader, String> NO_ITEMS = Collections.emptyMap();
+
+    /**
+     * Collects the traits from the given {@link HttpResponse}.
+     *
+     * @param httpResponse the HTTP reponse to collect traits from
+     * @param timestamp the UNIX timestamp when the response was received
+     * @return a new {@link Traits}
+     */
+    public static Traits collect(HttpResponse httpResponse, long timestamp) {
+        Builder<TraitHeader, String> builder = new ImmutableMap.Builder<TraitHeader, String>();
+
+        HeaderIterator headers = httpResponse.headerIterator();
+        while (headers.hasNext()) {
+            Header header = headers.nextHeader();
+            TraitHeader traitHeader = TraitHeader.fastValueOf(header.getName());
```

Comment:
Not sure the overhead of running this through the streaming API is worth it given there is no more than a handful of headers in a request.